modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/JavaParserUtils.java

```diff
@@ -0,0 +1,385 @@
+    MuleVersion finalCalculatedMMV = calculatedMMV;
+    return operationLevelMMV.map(mmv -> {
+      if (mmv.priorTo(finalCalculatedMMV)) {
+        LOGGER
+            .warn(format("Calculated Min Mule Version is %s which is greater than the one set at the operation level %s. Overriding it.",
+                         finalCalculatedMMV, mmv));
+        return finalCalculatedMMV;
+      } else {
+        return mmv;
+      }
+    }).orElse(calculatedMMV);
+  }
+
+  private static MuleVersion calculateInterfacesMinMuleVersion(Type type) {
+    MuleVersion calculatedMMV = getEnforcedMinMuleVersion(type);
+    if (belongsToSdkPackages(type.getTypeName())) {
+      return calculatedMMV;
+    }
+    for (Type interfaceType : type.getImplementingInterfaces()) {
+      calculatedMMV = maxMMV(calculatedMMV, calculateInterfacesMinMuleVersion(interfaceType));
+    }
+    return calculatedMMV;
+  }
+
+  private static MuleVersion calculateSourceGenericsMinMuleVersion(List<Type> sourceGenerics, String sourceName) {
+    if (sourceGenerics.size() != 2) {
+      throw new IllegalArgumentException(format("Source %s does not have 2 generics", sourceName));
+    }
+    Type outputType = sourceGenerics.get(0);
+    Type attributesType = sourceGenerics.get(1);
+    return maxMMV(getEnforcedMinMuleVersion(attributesType), calculateOutputMinMuleVersion(outputType));
+  }
+
+  private static MuleVersion calculateOutputMinMuleVersion(Type outputType) {
+    MuleVersion calculatedMMV = FIRST_MULE_VERSION;
+    if (outputType.asMetadataType() instanceof ArrayType) {
+      for (TypeGeneric typeGeneric : outputType.getGenerics()) {
+        calculatedMMV = maxMMV(calculatedMMV, calculateOutputMinMuleVersion(typeGeneric.getConcreteType()));
+      }
+    } else if (outputType.isSameType(Result.class) || outputType.isSameType(org.mule.sdk.api.runtime.operation.Result.class)) {
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(outputType));
+      for (TypeGeneric resultTypeGeneric : outputType.getGenerics()) {
+        calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(resultTypeGeneric.getConcreteType()));
+      }
+    } else if (outputType.isSameType(PagingProvider.class)
+        || outputType.isSameType(org.mule.sdk.api.runtime.streaming.PagingProvider.class)) {
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(outputType));
+      calculatedMMV = maxMMV(calculatedMMV, getEnforcedMinMuleVersion(outputType.getGenerics().get(1).getConcreteType()));
+    } else {
+      calculatedMMV = getEnforcedMinMuleVersion(outputType);
+    }
+    return calculatedMMV;
+  }
+
+  private static MuleVersion calculateFieldMinMuleVersion(ExtensionParameter field) {
+    Type parameterType = field.getType();
+    MuleVersion calculatedMMV = FIRST_MULE_VERSION;
+    for (Type annotation : field.getAnnotations()) {
+      if (annotation.isSameType(MinMuleVersion.class)) {
```

Comment:
Suppose the following example:
```
@Parameter
@MinMuleVersion(4.6)
String someField;
```
When moving from 4.5 to 4.6 the app would break as now there is a mandatory parameter.
So why would we allow it? What use case do we want to support?

I do agree that the validation should happen elsewhere, so removing it for now.